org.eclipse.jdt.core/batch/org/eclipse/jdt/internal/compiler/batch/ClasspathJep247Jdk12.java

```diff
@@ -122,14 +125,22 @@ public void initialize() throws IOException {
 			return;
 		}
 		URI uri = URI.create("jar:file:" + t.getRawPath()); //$NON-NLS-1$
-		try {
-			this.fs = FileSystems.getFileSystem(uri);
-		} catch(FileSystemNotFoundException fne) {
-			// Ignore and move on
-		}
+		this.fs = FILE_SYSTEM_CACHE.compute(uri, (old, u) -> {
+			try {
+				return FileSystems.getFileSystem(uri);
+			} catch (FileSystemNotFoundException fne) {
+				// create it
+				HashMap<String, ?> env = new HashMap<>();
+				try {
+					return FileSystems.newFileSystem(uri, env);
+				} catch (IOException e) {
+					// can't do anything useful then, will produce error below
+					return null;
```

Comment:
One should catch java.nio.file.FileSystemAlreadyExistsException and return getFileSystem(). 

Also would be good to have similar error handling like in https://github.com/eclipse-jdt/eclipse.jdt.core/pull/192 and not just silently hide original errors. Exception below doesn't provide the root cause.

Reply:
note: there are few places where similar problem with newFileSystem exists, one should go over all of them

Reply:
See constructor of CtSym, this can be also done without a map, JDK knows already created file systems.

Reply:
This FileSystems API is really a bit messy... According to the docs, even if the FS exits, it can throw exception because it was closed but that's provider specific ... So actually anything bad can happen any time, I actually wonder is there any advantage in using `FileSystem` here at all? Why not simply ZipFile?